### ISSUE TITLE
[CI] Remove `tim-actions/dco` action in favour of DCO app

### ISSUE THREAD

**Issue by feltech** [TheFoundryVisionmongers/otio-openassetio#11]
# What

Once the DCO app is approved and "installed", remove usages of `tim-actions/dco` ("Check messages" job)

# Why

The third-party action we were using is throwing up deprecation warnings and needs to be replaced. There is a handy GitHub app that is much better - see https://github.com/OpenAssetIO/OpenAssetIO/issues/709#issuecomment-1306342203

**Comment by foundrytom** [comment]
Closing in favour of the umbrella https://github.com/OpenAssetIO/OpenAssetIO/issues/731
